src/ParserFunctions/SlotTemplatesParserFunction.php

```diff
@@ -4,65 +4,79 @@
 
 use ComplexArray;
 use Error;
+use MediaWiki\MediaWikiServices;
 use MWException;
 use Parser;
 use TextContent;
+use WikibaseSolutions\MediaWikiTemplateParser\Parser as DeprecatedParser;
 use WikibaseSolutions\MediaWikiTemplateParser\RecursiveParser;
 use WSSlots\WikiPageTrait;
 use WSSlots\WSSlots;
-use WikibaseSolutions\MediaWikiTemplateParser\Parser as DeprecatedParser;
 
 /**
  * Handles the #slottemplates parser function.
  */
 class SlotTemplatesParserFunction {
-    use WikiPageTrait;
+	use WikiPageTrait;
+
+	/**
+	 * Execute the parser function.
+	 *
+	 * @param Parser $parser
+	 * @param string $slotName
+	 * @param string|null $pageName
+	 * @param string|null $arrayName
+	 * @param string|null $recursive
+	 * @return string
+	 * @throws MWException
+	 */
+	public function execute( Parser $parser, string $slotName, string $pageName = null, string $arrayName = null, string $recursive = null ): string {
```

Comment:
Exceeds 120 chars use 
```
public function execute(
		Parser $parser,
		string $slotName,
		string $pageName = null,
		string $arrayName = null,
		string $recursive = null
	) : string {
```

Reply:
Didn't see any weird things, except for some mw codestyling mentioned. Why not use the userCan function, as it does the check for you ?

Reply:
Thanks. The code now uses `userCan`.